ksqldb-functional-tests/src/test/resources/sql-tests/query-upgrades/filters.sql

```diff
@@ -0,0 +1,191 @@
+-- this file tests adding/removing/changing filters
+
+----------------------------------------------------------------------------------------------------
+--@test: add filter to basic STREAM without filter
+----------------------------------------------------------------------------------------------------
+SET 'ksql.create.or.replace.enabled' = 'true';
+
+CREATE STREAM a (id INT KEY, col1 INT) WITH (kafka_topic='a', value_format='JSON');
+CREATE STREAM b AS SELECT * FROM a;
+
+INSERT INTO a (id, col1) VALUES (1, 0);
+ASSERT VALUES b (id, col1) VALUES (1, 0);
+
+CREATE OR REPLACE STREAM b AS SELECT * FROM a WHERE col1 > 0;
+
+INSERT INTO a (id, col1) VALUES (1, 0);
+INSERT INTO a (id, col1) VALUES (1, 1);
+
+ASSERT VALUES b (id, col1) VALUES (1, 1);
+
+----------------------------------------------------------------------------------------------------
+--@test: remove filter from basic STREAM with filter
+----------------------------------------------------------------------------------------------------
+SET 'ksql.create.or.replace.enabled' = 'true';
+
+CREATE STREAM a (id INT KEY, col1 INT) WITH (kafka_topic='a', value_format='JSON');
+CREATE STREAM b AS SELECT * FROM a WHERE col1 > 0;
+
+INSERT INTO a (id, col1) VALUES (2, 0);
+
+CREATE OR REPLACE STREAM b AS SELECT * FROM a;
+
+INSERT INTO a (id, col1) VALUES (1, 0);
+
+ASSERT VALUES b (id, col1) VALUES (1, 0);
+
+----------------------------------------------------------------------------------------------------
+--@test: modify filter from basic STREAM with filter
+----------------------------------------------------------------------------------------------------
+SET 'ksql.create.or.replace.enabled' = 'true';
+
+CREATE STREAM a (id INT KEY, col1 INT) WITH (kafka_topic='a', value_format='JSON');
+CREATE STREAM b AS SELECT * FROM a WHERE col1 > 0;
+
+INSERT INTO a (id, col1) VALUES (1, 0);
+INSERT INTO a (id, col1) VALUES (1, 1);
+
+CREATE OR REPLACE STREAM b AS SELECT * FROM a WHERE col1 < 0;
+
+INSERT INTO a (id, col1) VALUES (1, 0);
+INSERT INTO a (id, col1) VALUES (1, -1);
+
+ASSERT VALUES b (id, col1) VALUES (1, 1);
+ASSERT VALUES b (id, col1) VALUES (1, -1);
+
+----------------------------------------------------------------------------------------------------
+-- note that each insert that is filtered out corresponds with a tombstone emitted into
+-- the topic
+--@test: add filter to basic TABLE without filter
+----------------------------------------------------------------------------------------------------
+SET 'ksql.create.or.replace.enabled' = 'true';
+
+CREATE TABLE a (id INT PRIMARY KEY, col1 INT) WITH (kafka_topic='a', value_format='JSON');
+CREATE TABLE b AS SELECT * FROM a;
+
+INSERT INTO a (id, col1) VALUES (1, 0);
+ASSERT VALUES b (id, col1) VALUES (1, 0);
```

Comment:
The Assert consumes from the topic that's why after the replace you only have one record in the input topic with `(1,0)` ?